### PR TITLE
fix colLabels in multiple scenarios

### PR DESCRIPTION
**Pull request by patrickweh**
add placeholder for lazy use

add migration for layout storage (persistant column layout)

add ability to set wildcard to availableCols

add translated labels for operators

remove skipRender

add functions for availableCols and enabledCols

add aggregate labels

add ability to reset a layout

### CODE REVIEW

File: src/DataTable.php [hunk 1]
```diff
@@ -274,28 +283,31 @@ public function mount(): void
...
-        $this->availableCols = array_values(
-            array_unique(
-                array_merge(
-                    $this->enabledCols,
-                    $this->availableCols,
-                    [$this->modelKeyName]
-                )
-            )
-        );
+        $this->loadFilter($loadFilter ?? [], false);
```
Comment (SirSplasch) [inline src/DataTable.php:307]: can this not be ?:

File: src/DataTable.php [hunk 2]
```diff
@@ -307,6 +319,8 @@ public function boot(): void
 
     public function render(): View|Factory|Application
     {
+        $this->forceRender = false;
```
Comment (SirSplasch) [inline src/DataTable.php:322]: This means if one wants to set forceRender = true, the render() method needs to be overwritten.

File: src/DataTable.php [hunk 3]
```diff
@@ -1063,6 +1150,45 @@ private function cacheState(): void
...
+                ->save();
+        } catch (MissingTraitException) {
+        }
+    }
+
+    public function resetLayout(): void
+    {
+        try {
+            $this->ensureAuthHasTrait();
+            $layout = DatatableUserSetting::query()
```
Comment (SirSplasch) [inline src/DataTable.php:1180]: isn't there a where('component', $this->getCacheKey()) missing?

File: src/DataTable.php [hunk 4]
```diff
@@ -778,7 +852,7 @@ public function loadRelations(): array
             })
             ->map(function (Relation $relation) use ($basis) {
                 return [
-                    'value' => $relation->name,
+                    'value' => Str::snake($relation->name),
```
Comment (SirSplasch) [inline src/DataTable.php:855]: Isn't this a gamble? I mean, a relation name does not necessarily mean that the table is named the same only in snake_case.

**Review by SirSplasch — CHANGES_REQUESTED** [review]

File: src/DataTable.php (hunk 1 above)
Comment (patrickweh) [reply src/DataTable.php:307]: $a = [] :? "B"

This would be "B"

File: src/DataTable.php (hunk 2 above)
Comment (patrickweh) [reply src/DataTable.php:322]: intended, forceRender is like skiprender, only valid per request…

if you have a function that should trigger a render you have to set forceRender in this function to true.

File: src/DataTable.php (hunk 4 above)
Comment (patrickweh) [reply src/DataTable.php:855]: when doing a ->toArray() all relation keys are snake case, no matter if they are camel or snake in the class.

**Review by SirSplasch — CHANGES_REQUESTED** [review]

**Pull request merged**